src/compute_horde_prompt_gen/run.py

```diff
@@ -38,22 +42,28 @@ def generate_prompts(
         )
 
         seconds_taken = (datetime.datetime.now() - start_ts).total_seconds()
-        log.info(f"{i=} generation took {seconds_taken:.2f}s")
 
         new_prompts = []
         for j, sequence in enumerate(sequences):
             generated_prompts = parse_output(sequence)
-            log.debug(f"{i=} sequence={j} {generated_prompts=} from {sequence=}")
-
-            log.info(f"{i=} sequence={j} generated {len(generated_prompts)} prompts")
             new_prompts.extend(generated_prompts)
 
         # check_prompts_quality(new_prompts)
 
         # remove any duplicates
         new_prompts = list(set(new_prompts))
+        log.info(
+            f"{i=} generation took {seconds_taken:.2f}s; generated {len(new_prompts)} prompts"
+        )
 
-        if total_prompts - len(new_prompts) < 0:
+        # Use leftover prompts from previous batch if available
+        while leftover_prompts and total_prompts > 0:
+            new_prompts.append(leftover_prompts.popleft())
+            total_prompts -= 1
+
+        if len(new_prompts) > total_prompts:
+            # Save extra prompts for next batch
+            leftover_prompts.extend(new_prompts[total_prompts:])
```

Comment:
I think you need the original value of `total_prompts` that was passed in the function. We are subtracting the number of prompts generated from this variable.